src/coreclr/jit/emit.cpp

```diff
@@ -4102,6 +4117,213 @@ void emitter::emitDispCommentForHandle(size_t handle, GenTreeFlags flag)
 #endif // DEBUG
 }
 
+//****************************************************************************
+// emitRemoveJumpToNextInst:  Checks all jumps in the jump list to see if they are
+//   unconditional jumps generated at the end of BB_ALWAYS basic blocks and whether
+//   they are now a jump to the next instruction, this can happen when the following
+//   basic block emitted no instructions
+//
+// Returns:
+//   true if any jumps have been removed, otherwise returns false
+//
+// Assumptions:
+//    the jump list must be ordered by increasing igNum+insNo
+//
+void emitter::emitRemoveJumpToNextInst()
+{
+#ifdef TARGET_XARCH
```

Comment:
Why is this xarch specific?

Reply:
Because the caller is in shared code, codegenlinear, and the implementation relies on the new instrDescJmp field which is xarch only. it's cleaner to have no implementation and allow the compiler to optimize it away.